### DISCUSSION TITLE
Passing arguments to Mixins

### DISCUSSION THREAD

**Discussion by everdrone** [ent/ent#2763] [Q&A] [answered]
Hi, I'm pretty new to Ent and Go in general so this might be a noob question.

I'm trying to pass a parameter to a custom ID generator mixin.

The idea is to pass a string prefix to differentiate models in my database. so a `Post.ID` would look like `"post_abc0123"` and a `User.ID` would be `"user_abc0123"`.

Following [the documentation](https://entgo.io/docs/faq#how-to-use-a-custom-generator-of-ids), how would you pass a string argument to `func IDHook()` when declaring a model struct?

```go
func (Post) Mixin() []ent.Mixin {
	return []ent.Mixin{
		CustomIDMixin{
			Prefix: "post_",
		},
		TimeMixin{},
	}
}
```

**Comment by a8m** [comment, marked as answer]
Hey @everdrone 👋 
I think [this](https://github.com/ent/contrib/blob/master/entgql/internal/todopulid/ent/schema/pulid/mixin.go#L25-L27) example may help you. See also its usage [here](https://github.com/ent/contrib/blob/master/entgql/internal/todopulid/ent/schema/todo.go#L33). 

You're welcome to join our [Discord server](https://discord.gg/qZmPgTE6RX) if you still need help with this.

**Reply by everdrone** [reply]
Awesome! Thanks!
